Dockerfile

```diff
@@ -1,11 +1,10 @@
 FROM ubuntu:trusty
 ADD . /code
 WORKDIR /code
-RUN apt-get update
-RUN apt-get install -y git software-properties-common python python-dev python-pip
-RUN add-apt-repository -y ppa:chris-lea/node.js
-RUN apt-get update
-RUN apt-get install -y nodejs
-RUN npm install bower
-RUN pip install -r requirements.txt
+ENV DEBIAN_FRONTEND noninteractive
+RUN add-apt-repository -y ppa:chris-lea/node.js && \
```

Comment:
Yes, it is good to join lines to minimise the number of Docker image levels, but `add-apt-repository` does not exist if you don't do `update` first...